domain/network/state/subnet.go

```diff
@@ -80,103 +102,121 @@ VALUES (?, ?)`
 	insertAvailabilityZoneSubnetStmt := `
 INSERT INTO availability_zone_subnet (availability_zone_uuid, subnet_uuid)
 VALUES (?, ?)`
-	err = db.StdTxn(ctx, func(ctx context.Context, tx *sql.Tx) error {
-		// Add the subnet entity.
-		if _, err := tx.ExecContext(
-			ctx,
-			insertSubnetStmt,
-			uuid.String(),
-			cidr,
-			VLANTag,
-			spaceUUIDValue,
-			subnetType,
-		); err != nil {
-			return errors.Trace(err)
-		}
 
-		if subnetType == subnetTypeFanOverlaySegment {
-			// Retrieve the underlay subnet uuid.
-			var underlaySubnetUUID string
-			row := tx.QueryRowContext(ctx, retrieveUnderlaySubnetUUIDStmt, fanInfo.FanLocalUnderlay)
-			if err := row.Scan(&underlaySubnetUUID); err != nil {
-				return errors.Trace(err)
-			}
-			// Add the association of the underlay and the newly
-			// created subnet to the associations table.
-			if _, err := tx.ExecContext(
-				ctx,
-				insertSubnetAssociationStmt,
-				uuid.String(),
-				underlaySubnetUUID,
-			); err != nil {
-				return errors.Trace(err)
-			}
-		}
-		// Add the subnet uuid to the provider ids table.
-		if _, err := tx.ExecContext(
-			ctx,
-			insertSubnetProviderIDStmt,
-			providerID,
-			uuid.String(),
-		); err != nil {
+	// Add the subnet entity.
+	if _, err := tx.ExecContext(
+		ctx,
+		insertSubnetStmt,
+		subnet.ID,
+		subnet.CIDR,
+		subnet.VLANTag,
+		spaceUUIDValue,
+		subnetType,
+	); err != nil {
+		return errors.Trace(err)
+	}
+
+	if subnetType == subnetTypeFanOverlaySegment {
+		// Retrieve the underlay subnet uuid.
+		var underlaySubnetUUID string
+		row := tx.QueryRowContext(ctx, retrieveUnderlaySubnetUUIDStmt, subnet.FanInfo.FanLocalUnderlay)
+		if err := row.Scan(&underlaySubnetUUID); err != nil {
 			return errors.Trace(err)
```

Comment:
Can we annotate the errors in this function, it's going to be hard to know which one fails easily.

Reply:
sure, I thought I had already done that